src/peers/component.py

```diff
@@ -0,0 +1,46 @@
+import json
+from typing import List, Dict
+from opus import Asset
+from util.event_emitter import EventEmitter
+
+from websockets.server import WebSocketServerProtocol
+
+
+class ComponentPeer(EventEmitter):
+    """
+    Base class for peers which accepts actions.
+
+    Parameters
+    ----------
+    target_types
+        Target types this peer listens too (e.g. audio, video, image etc.)
+    """
+
+    def __init__(self, target_types: List[str]):
+        self._target_types = target_types
+        self._websockets: List[WebSocketServerProtocol] = []
+
+    async def handle_socket(self, websocket: WebSocketServerProtocol):
```

Comment:
Could there be component-specific messages we need to handle in the future? If so, it could be a problem to define this method here.

I think having it in each subclass is more extensible, even if there is some duplicated code.